src/QuantumSdk/DefaultItems/DefaultItems.targets

```diff
@@ -59,8 +59,9 @@
   <PropertyGroup>
     <DefaultEntryPointGeneration>dotnet "$(MSBuildThisFileDirectory)../tools/utils/Microsoft.Quantum.Sdk.DefaultEntryPoint.Generation.dll"</DefaultEntryPointGeneration>
     <!-- true/false optional properties -->
-    <QirGeneration Condition="'$(QirOutputPath)' != '' And '$(QirGeneration)' == 'default'">true</QirGeneration>
-    <QirGeneration Condition="'$(QirOutputPath)' == '' And '$(QirGeneration)' == 'default'">false</QirGeneration>
+    <Warning Condition="'$(QirGeneration)' == 'true' And '$(ResolvedQSharpOutputType)' == 'QSharpLibrary'" Text="QIR Generation 'true' is not supported on library projects, returning to default behavior." />
+    <QirGeneration Condition="'$(QirOutputPath)' != '' And '$(QirGeneration)' == 'default' And '$(ResolvedQSharpOutputType)' == 'QSharpExe'">true</QirGeneration>
+    <QirGeneration Condition="('$(QirOutputPath)' == '' And '$(QirGeneration)' == 'default') Or '$(ResolvedQSharpOutputType)' == 'QSharpLibrary'">false</QirGeneration>
```

Comment:
Would it be safer to have `'$(ResolvedQSharpOutputType)' != 'QSharpExe'` instead of `'$(ResolvedQSharpOutputType)' == 'QSharpLibrary'`?